CONTRIBUTING.md

```diff
@@ -165,3 +165,70 @@ general guidelines for the community as a whole:
   respond to an email you are potentially sending to a large number of
   people.  Please consider this before you update.  Also remember that
   nobody likes spam.
+
+## FAQ
+
+**Is it fine to go with Kotlin for my contribution?**
+
+We don't use Kotlin. This is due to the fact, that our server components
+are written in pure Java. Even tough the Android community is adapting
+Kotlin well, most guys from the Java community are not used to writing
+Kotlin code before.
+
+**Is there an architectural design for the presentation layer?**
+
+There is no MVVM/MVP or MVC in place, because the app is not too complex
+and it would not really benefit from something like that right now.
+
+**Is there an architectural design for the data layer?**
+
+The data layer is separated to it's own package.
+
+**Where does the Sitemap rendering happen?**
+
+WidgetListFragment + (especially) WidgetAdapter
+
+**How is navigation managed?**
+
+ContentController manages how fragments are displayed, and MainActivity 
+commands it what fragments to display. Navigation is too dynamic in our
+app (determined by Sitemap) for the navigation architecture component to
+be useful.
+
+**Do we use fragments in our activities? If yes, are they necessary?**
+
+The answer to the first question is 'it depends'.
+They should be used whenever it makes sense to use them.
```

Comment:
For very simple activities, e.g. LogActivity, we don't use fragments, otherwise we do.